samcli/local/docker/manager.py

```diff
@@ -76,13 +86,14 @@ def pull_image(self, image_name, stream=None):
 
         :param string image_name: Name of the image
         :param stream: Optional stream to write output to. Defaults to stderr
-        :raises DockerImageNotFoundException: If the Docker image was not available in the server
+        :raises DockerImagePullFailedException: If the Docker image was not available in the server
         """
         stream = stream or sys.stderr
         try:
             result_itr = self.docker_client.api.pull(image_name, stream=True, decode=True)
-        except docker.errors.ImageNotFound as ex:
-            raise DockerImageNotFoundException(str(ex))
+        except docker.errors.APIError as ex:
```

Comment:
[nit] Looks like the default timeout on Docker Client is 60 seconds. Is it possible to add timeout to the `docker_client.api.pull` call as well that won't hog for 60 seconds if it can't find a network?

Reply:
Good callout. I don't see this as an option for in `docker_client.api.pull` but this is configurable in the `docker_client`. This will affect poor network instead of no network (I have been testing locally with wifi off without waiting for the timeout). I can update the `docker_client` we use but what would be a reasonable timeout for this?

This PR doesn't handle the poor network case and I didn't aim to solve this here.

Reply:
It’s a good practice to set timeouts anyway. In this case we can be more aggressive. If you want to fix poor network later, I am okay.

Reply:
Agree. I want to tackle the greater poor network connectivity problem separately from this PR.